setup.sh

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env bash
+set -e
+
+readonly SYSTEM=$(uname -s)
+EXTRA_STEPS=()
+
+linux() {
+  [[ ${SYSTEM} == 'Linux' ]]
+}
+
+mac() {
+  [[ ${SYSTEM} == 'Darwin' ]]
+}
+
+installed() {
+  hash "$1" 2>/dev/null
+}
+
+nvm_installed() {
+  if [ -d '/usr/local/opt/nvm' ] || [ -d "$HOME/.nvm" ]; then
+    true
+  else
+    false
+  fi
+}
+
+nvm_available() {
+  type -t nvm > /dev/null
+}
+
+source_nvm() {
+  if ! nvm_available; then
+    [ -e "/usr/local/opt/nvm/nvm.sh" ] && source /usr/local/opt/nvm/nvm.sh
+  fi
+  if ! nvm_available; then
+    [ -e "$HOME/.nvm/nvm.sh" ] && source $HOME/.nvm/nvm.sh
+  fi
+}
+
+check_encryption() {
+  if linux; then
+    EXTRA_STEPS+=("Sorry, can't check if your hard disk is encrypted - please ensure that it is! (applies to both portable and Desktop machines)")
+  elif mac; then
+    if [[ "$(fdesetup status)" != "FileVault is On." ]]; then
+      EXTRA_STEPS+=("your hard disk is not encrypted! Encryption must be enabled on all guardian machines. Follow these instructions: https://support.apple.com/en-gb/HT204837")
+    fi
+  fi
+}
+
+create_aws_config() {
+  local path="$HOME/.aws"
+  local filename="config"
+
+  if [[ ! -f "$path/$filename" ]]; then
+    if [[ ! -d "$path" ]]; then
+      mkdir "$path"
+    fi
+
+    echo "[profile membership]
+region = eu-west-1" > "$path/$filename"
+  fi
+}
+
+install_homebrew() {
+  if mac && ! installed brew; then
+    ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+  fi
+}
+
+install_jdk() {
+  if ! installed javac; then
+    if linux; then
+      sudo apt-get install -y openjdk-8-jdk
+    elif mac; then
+      EXTRA_STEPS+=("Download the JDK from http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html")
```

Comment:
We should not encourage people to install Oracle JDK, but `OpenJDK`!
People should use `https://adoptopenjdk.net` for instance but not `oracle` website.